clusterloader2/pkg/measurement/common/probes/probes.go

```diff
@@ -157,54 +160,50 @@ func (p *probesMeasurement) start(config *measurement.MeasurementConfig) error {
 	return nil
 }
 
-func (p *probesMeasurement) gather(params map[string]interface{}) ([]measurement.Summary, error) {
+func (p *probesMeasurement) gather(params map[string]interface{}) (measurement.Summary, error) {
 	klog.Info("Gathering metrics from probes...")
 	if p.startTime.IsZero() {
 		return nil, fmt.Errorf("measurement %s has not been started", p)
 	}
-	thresholds, err := parseThresholds(params)
+	threshold, err := util.GetDurationOrDefault(params, "threshold", 0)
 	if err != nil {
 		return nil, err
 	}
 	measurementEnd := time.Now()
-	var probeSummaries []measurement.Summary
-	var violationErrors []error
-	for probeName, queryTmpl := range p.probeNameToPrometheusQueryTmpl {
-		query := prepareQuery(queryTmpl, p.startTime, measurementEnd)
-		executor := measurementutil.NewQueryExecutor(p.framework.GetClientSets().GetClient())
-		samples, err := executor.Query(query, measurementEnd)
-		if err != nil {
-			return nil, err
-		}
 
-		latencyMetric, err := measurementutil.NewLatencyMetricPrometheus(samples)
-		if err != nil {
-			return nil, err
-		}
-		prefix, suffix := "", ""
-		if threshold, ok := thresholds[probeName]; ok {
-			suffix = fmt.Sprintf(", expected perc99 <= %v", threshold.Perc99)
-			if err = latencyMetric.VerifyThreshold(threshold); err != nil {
-				violationErrors = append(violationErrors, errors.NewMetricViolationError(probeName, err.Error()))
-				prefix = " WARNING"
-			}
-		}
-		klog.Infof("%s:%s got %v%s", probeName, prefix, latencyMetric, suffix)
+	query := prepareQuery(p.config.Query, p.startTime, measurementEnd)
+	executor := measurementutil.NewQueryExecutor(p.framework.GetClientSets().GetClient())
+	samples, err := executor.Query(query, measurementEnd)
+	if err != nil {
+		return nil, err
+	}
 
-		probeSummary, err := createSummary(probeName, *latencyMetric)
-		if err != nil {
-			return nil, err
+	latency, err := measurementutil.NewLatencyMetricPrometheus(samples)
+	if err != nil {
+		return nil, err
+	}
+
+	var violation error
+	prefix, suffix := "", ""
+	if threshold > 0 {
+		suffix = fmt.Sprintf(", expected perc99 <= %v", threshold)
+		if latency.Perc99 > threshold {
```

Comment:
I actually liked the previous VerifyThreshold - why can't we create and use it here too?

Reply:
I didn't like it - it was implicit and unnecessarily complicated for what can be a simple check of two floats.

Reply:
Hmm - I actually think that we may take advantage of it, and do comparisons on smaller percentiles too.